syntax = "proto3";

package hello;

// Hello World服务定义
service HelloService {
  // 发送问候消息
  rpc SayHello (HelloRequest) returns (HelloReply) {}
  
  // 发送多个问候消息（服务器流）
  rpc SayHelloStream (HelloRequest) returns (stream HelloReply) {}
}

// 请求消息
message HelloRequest {
  string name = 1;
}

// 响应消息
message HelloReply {
  string message = 1;
}